core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -206,7 +206,8 @@ public CarbonTable getTable() {
       Set<Path> partitionLocations, List<ExtendedBlocklet> blocklets,
       Map<Segment, List<Index>> indexes) throws IOException {
     for (Segment segment : segments) {
-      if (indexes.get(segment).isEmpty() || indexes.get(segment) == null) {
+      if (segment == null ||
+          indexes.get(segment).isEmpty() || indexes.get(segment) == null) {
```

Comment:
change to:
segment == null || indexes.get(segment) == null || indexes.get(segment).isEmpty()

Reply:
modified